### PR TITLE
Tidy C_TO_KELVIN / create KELVIN_TO_C

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Review by khancyr — APPROVED** [review]
I am not really fan of macros, but why not

File: Tools/AP_Periph/can.cpp [hunk 1]
```diff
@@ -1737,7 +1737,7 @@ void AP_Periph_FW::can_battery_update(void)
         if (battery.lib.get_temperature(temperature, i)) {
             // Battery lib reports temperature in Celsius.
             // Convert Celsius to Kelvin for tranmission on CAN.
```
Comment (RuffaloLavoisier) [inline Tools/AP_Periph/can.cpp:1739]: Thank you for changing it to `tranmission` -> `transmission`.

File: libraries/AP_EFI/AP_EFI_NWPMU.cpp [hunk 2]
```diff
@@ -86,12 +86,12 @@ void AP_EFI_NWPMU::handle_frame(AP_HAL::CANFrame &frame)
...
-            internal_state.coolant_temperature = data.coolant_temp * 0.1f + C_TO_KELVIN;
-            internal_state.cylinder_status[0].cylinder_head_temperature = data.coolant_temp * 0.1f + C_TO_KELVIN;
+            internal_state.coolant_temperature = C_TO_KELVIN(data.coolant_temp * 0.1f);
+            internal_state.cylinder_status[0].cylinder_head_temperature = C_TO_KELVIN(data.coolant_temp * 0.1f);
             break;
         case NWPMU_TEMPERATURE_TYPE::F:
-            internal_state.coolant_temperature = ((data.coolant_temp * 0.1f) - 32 * 5/9) + C_TO_KELVIN;
-            internal_state.cylinder_status[0].cylinder_head_temperature = ((data.coolant_temp * 0.1f) - 32 * 5/9) + C_TO_KELVIN;
+            internal_state.coolant_temperature = C_TO_KELVIN(((data.coolant_temp * 0.1f) - 32 * 5/9));
+            internal_state.cylinder_status[0].cylinder_head_temperature = C_TO_KELVIN(((data.coolant_temp * 0.1f) - 32 * 5/9));
```
Comment (hendjoshsr71) [inline libraries/AP_EFI/AP_EFI_NWPMU.cpp:94]: HAHA could use a F_TO_KELVIN here instead ;)

**Review by hendjoshsr71 — APPROVED** [review]
LGTM besides a nitpick

Once you fixup the small error causing tests to fail.

File: Tools/AP_Periph/can.cpp (hunk 1 above)
Comment (peterbarker) [reply Tools/AP_Periph/can.cpp]: Done, thanks

File: libraries/AP_EFI/AP_EFI_NWPMU.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_EFI/AP_EFI_NWPMU.cpp]: ... and while we're at it - we can make it *not wrong* :-)  Seriously whoops.

**Review by hendjoshsr71 — APPROVED** [review]
LGTM

File: libraries/AP_EFI/AP_EFI_NWPMU.cpp (hunk 2 above)
Comment (hendjoshsr71) [reply libraries/AP_EFI/AP_EFI_NWPMU.cpp]: Good catch I didn't notice the subtle parenthesis error there ..... just that it was F to K

**Comment by peterbarker** [comment]
Oh... I should mention this does a drive-by fix of a f-to-kelvin conversion.

**Pull request merged**